jobrunner/job_executor.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Protocol, Mapping, List, Tuple, Optional
+
+from jobrunner.models import State, StatusCode
+
+
+class Privacy(Enum):
+    HIGH = "high"
+    MEDIUM = "medium"
+
+
+InputSpec = Mapping[str, str]  # file paths to actions that produced them
+OutputSpec = Mapping[str, Mapping[str, str]]  # privacy levels to names to patterns
```

Comment:
I think we can simplify `InputSpec` to just a list of files (unless I'm missing some reason we need the action too). And `OutputSpec` can just be a mapping of patterns to privacy levels.

But maybe you're just trying to capture the types of the current status quo before refactoring?

Reply:
Yes, you're right twice. :-) The actions from the input spec are logged at the moment, but I think we can live without that.